### ISSUE TITLE
Android Problem

### ISSUE THREAD

**Issue by AndreKoepke** [pi-hole/pi-hole#1235]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*
- [x] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

_1_

---

**[BUG | ISSUE] Expected Behaviour:**
Pi-Hole working as DNS which blocks all ad-Domains. For that, Pi-Hole must be the only used DNS-Server for all devices.

**[BUG | ISSUE] Actual Behaviour:**
Android is tricky. It uses the gateway as 3rd DNS-server. (Confirmed on Android 6.0.1.)
In my case, im not able to configure the WAN-DNS in my router (bad provider ...), so all ads will blocked by Pi-Hole in first DNS-Lookup but Android starts other DNS-requests -> Ads still there.

**[BUG | ISSUE] Steps to reproduce:**

- Set the DNS-WAN router to 8.8.8.8
- Android-Devices will now showing ads

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**Comment by DL6ER** [comment]
I have an Android 6.0.1 device. Can you provide further details what I'd have to set up where exactly in order to be able to reproduce? I use the Pi-hole DHCP server in my local network (IPv4 + IPv6) but obviously my gateway is still my router (in which I cannot change the upstream DNS as well). However, I don't see any issues.

**Comment by AndreKoepke** [comment]
# My setup
- Router is a `HomeBox 6441` (local IP is `192.168.22.1`)
- Pi-Hole's IP is `192.168.22.75`
- Android 6.0.1 device is only connected by WLAN
- Gateway is the router, DNS1 is Pi-Hole, DNS2 is nulled
- Ads still visible, Pi-Hole's log show it was blocked

I readed this [post](https://discourse.pi-hole.net/t/android-ignores-dns-solved/457/5).
To confirm this, I set DNS1 and DNS2 to `0.0.0.0` and internet ist working well.

**Comment by DL6ER** [comment]
> Gateway is the router, DNS1 is Pi-Hole, DNS2 is nulled

How did you set these settings on your Android device?

**Comment by AndreKoepke** [comment]
Normal WLAN-settings (german device).
![2017-02-10 10_52_42-telegramdesktop](https://cloud.githubusercontent.com/assets/15613049/22822293/1f0d4f8c-ef7f-11e6-9524-9660fbf08891.png)

**Comment by DL6ER** [comment]
Okay, static configuration - that was the bit I was missing in your description. I'll try that when I'm at home.
My Android devices are all running with DHCP configuration.

**Comment by AndreKoepke** [comment]
Sry. :)

But running over DHCP (with Pi-Hole as server) has same results.
![2017-02-10 10_52_42-telegramdesktop](https://cloud.githubusercontent.com/assets/15613049/22822669/a5a9d7da-ef80-11e6-8825-2b8cfb02efe2.png)

The android-device is the only one, which (however) bypass Pi-Hole.

**Comment by DL6ER** [comment]
Which App did you use when making this screenshot? Also, did I get this right that the problem (gateway is used as DNS server) is *not* apparent in this picture?

**Comment by AndreKoepke** [comment]
`Network Info II`

No, in this pic you can not see, that android uses the gateway as DNS.
It was only a "proof-pic" for correct settings.

**Comment by r0ckarong** [comment]
I had the same problem. For me it was that my Android 6 device apparently uses SLAAC to negotiate an IPv6 Address with the router directly and ignored my v6 DHCP settings (since Android does not actually support v6 DHCP from some internet stuff I read). 

I turned off an option in my router settings that allowed SLAAC across the network and now it works as desired with only IPv4 addresses). It's probably the same IPv6 leak that you're looking at.

http://www.techrepublic.com/article/androids-lack-of-dhcpv6-support-poses-security-and-ipv6-deployment-issues/

**Comment by darkrain-nl** [comment]
That is for sure possible, but it is also possible to configure it in such a way that it works...

You are correct that Android doesn't support DHCPv6, let me explain my setup, it might help others.
I use Pihole as DHCP server, both IPv4 and IPv6.
DHCP on my router is turned off, except for IPv6 I have automatic address assignment turned on. The type is SLAAC+stateless DHCP. This makes the router send Router Advertisements with the default route as payload. (needed for internet to work)

Pihole sends Router Advertisements with the IPv6 DNS server address as payload, but without default route. (Pihole is not a router, so this is correct)

This all combined makes IPv6 work on Windows, Linux, Android and so on... The default route comes from the router, the DNS server address comes from Pihole. 
Ads are blocked on both IPv4 and Ipv6 on all devices, I'm personally very happy with this setup.

**Comment by AndreKoepke** [comment]
Hm, Iam having not much settings for IPv6 in my router (only one radio-box with "IPv6-DHCP stateless"-label)

My solution is: I will buy a new router which dosn't has a buggy provider operating system.

**Comment by DL6ER** [comment]
> My solution is: I will buy a new router which dosn't has a buggy provider operating system.

I have another issue with a provider provided router and IPv6 (issue #1212). I might have to do the same...

**Comment by dschaper** [comment]
Is this still and ongoing issue if the addressing is changed to utilize ULAs on the local network?

**Comment by DictumMortuum** [comment]
I think that I have exactly the same issue.

gateway: 192.168.1.1
dns: 192.168.1.10 (pi-hole)

ads are blocked on desktop, but not on mobile devices.

**Comment by AndreKoepke** [comment]
Iam now using a Fritz!Box (it has a setting to change the DNS). It worked for me.
